vos/vos.py

```diff
@@ -127,6 +127,34 @@ def __str__(self):
                                                        self.netloc, self.path)
 
 
+class RetrySession(requests.Session):
+
+    def send(self, request, **kwargs):
+        """
+        Send a given PreparedRequest, wrapping the connection to service in try/except that retries on
+        Connection reset by peer.
+
+        :param request: The prepared request to send.
+        :param kwargs: Any keywords the adaptor for the request accepts.
+        :return: the response
+        :rtype: requests.Response
+        """
+
+        total_delay = 0
+        current_delay = DEFAULT_RETRY_DELAY
```

Comment:
I still think that this logic is more complicated than it should be. Is there anything wrong with specifying the initial DEFAULT_RETRY_DELAY and the number of retries? 
The loop will be on the number of retries and the current_delay *= 2. Simpler in my opinion.